### PR TITLE
Development preview features coverage

### PR DESCRIPTION
**Pull request by jeanchlopez**
Deleted old branch (4.7-devpreview-features) to easily fix my conflicts. This is the new branch with the same content.

### CODE REVIEW

File: training/modules/ocs4/pages/ocs4-additionalfeatures-dbwal.adoc [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        storageClassName: {storageclass}
+        volumeMode: Block
+----
+
+When added to your storage cluster the following will happen:
+
+. Each OSD data will claim a PVC as `\{deviceset-prefix-x-data-y\}`
+. Each OSD RockDB metadata will claim a PVC as `\{deviceset-prefix-x-metadata-y\}`
+
+CAUTION: The deployment process will provision PVC and will not be able to partition
```
Comment (netzzer) [inline training/modules/ocs4/pages/ocs4-additionalfeatures-dbwal.adoc:62]: If I understand this correctly I would change:
The deployment process will provision PVC and will not be able to partition
an existing device. Make sure to configure your RocksDB metadata partitions
before configuring LSO and before deploying your storage cluster.

To:
The LSO discovery and subsequent provisioning will create new PVs using the entire disk and you cannot create partitions after this process . Make sure to configure your RocksDB metadata disk partitions
before LSO discovery to create the data and metadata PVs to be used to deploy your storage cluster.

File: training/modules/ocs4/pages/ocs4-additionalfeatures-dbwal.adoc [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+----
+
+When added to your storage cluster the following will happen:
+
+. Each OSD data will claim a PVC as `\{deviceset-prefix-x-data-y\}`
+. Each OSD RockDB WAL will claim a PVC as `\{deviceset-prefix-x-wal-y\}`
+
+NOTE: The WAL placement can be combined with the RocksDB metadata placement.
+
+CAUTION: The deployment process will provision PVC and will not be able to partition
```
Comment (netzzer) [inline training/modules/ocs4/pages/ocs4-additionalfeatures-dbwal.adoc:93]: Same input as above with change to `wal`.

The LSO discovery and subsequent provisioning will create new PVs using the entire disk and you cannot create partitions after this process . Make sure to configure your RocksDB metadata disk partitions
before LSO discovery to create the data and wal PVs to be used to deploy your storage cluster.

File: training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc [hunk 3]
```diff
@@ -0,0 +1,125 @@
+=== Mixed OSD device type configuration
+
+With ODF 4.7 you can customize the deployment of your OSDs to consume different device
```
Comment (netzzer) [inline training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc:3]: ODF 4.7 and above?

File: training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+types. This feature can be combined with the BlueStore placement customization and
```
Comment (netzzer) [inline training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc:4]: replace 'BlueStore placement' with 'BlueStore RocksDB metadata and WAL placement'

File: training/modules/ocs4/pages/ocs4-additionalfeatures-override.adoc [hunk 5]
```diff
@@ -0,0 +1,41 @@
+=== Ceph configuration override
+
+ODF 4.7 allows you to create a custom configuration map containing Ceph configuration
```
Comment (netzzer) [inline training/modules/ocs4/pages/ocs4-additionalfeatures-override.adoc:3]: ODF 4.7 and above?

File: training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+  storageDeviceSets:
+  - count: 1
+    dataPVCTemplate:
+      spec:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: {size}
+        storageClassName: {storageclass}
```
Comment (netzzer) [inline training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc:28]: storageclass-nvme

File: training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+    deviceType: nvme <1>
+  - count: 1
+    dataPVCTemplate:
+      spec:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: {size}
+        storageClassName: {storageclass}
```
Comment (netzzer) [inline training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc:42]: storageclass-hdd

File: training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+        volumeMode: Block
+  - count: 1
+    dataPVCTemplate:
+      spec:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: {size}
+        storageClassName: {storageclass}
```
Comment (netzzer) [inline training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc:65]: storageclass-ssd

File: training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+    replica: 3
+    deviceType: ssd <3>
+    metadataPVCTemplate:
+      spec:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: {size}
+        storageClassName: {storageclass}
```
Comment (netzzer) [inline training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc:78]: what storageclass should this be?

File: training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+        volumeMode: Block
+    walPVCTemplate:
+      spec:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: {size}
+        storageClassName: {storageclass}
```
Comment (netzzer) [inline training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc:87]: what storageclass should this be?

File: training/modules/ocs4/pages/ocs4-additionalfeatures-override.adoc [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+[source,yaml]
+----
+spec:
+  managedResources:
+    cephConfig:
+      reconcileStrategy: ignore
+[...]
+----
+
+Once this parameter is added to your *StorageClkuster* CR you simply have to create
```
Comment (netzzer) [inline training/modules/ocs4/pages/ocs4-additionalfeatures-override.adoc:19]: In the openshift-storage namespace, meaning you need to create namespace before CM and then storagecluster created.

File: training/modules/ocs4/pages/ocs4-additionalfeatures-dbwal.adoc [hunk 12]
```diff
@@ -0,0 +1,95 @@
+=== OSD BlueStore RocksDB metadata and WAL placement
+
+With ODF 4.7 you can customized the deployment of your OSDs when it comes to what device to use
```
[the remark this answers is not in the document]
Comment (jeanchlopez) [reply training/modules/ocs4/pages/ocs4-additionalfeatures-dbwal.adoc:3]: Done

File: training/modules/ocs4/pages/ocs4-additionalfeatures-dbwal.adoc (hunk 1 above)
Comment (jeanchlopez) [reply training/modules/ocs4/pages/ocs4-additionalfeatures-dbwal.adoc:62]: Done

File: training/modules/ocs4/pages/ocs4-additionalfeatures-dbwal.adoc (hunk 2 above)
Comment (jeanchlopez) [reply training/modules/ocs4/pages/ocs4-additionalfeatures-dbwal.adoc:93]: Done

Comment (jeanchlopez) [reply training/modules/ocs4/pages/ocs4-additionalfeatures-dbwal.adoc:93]: Done

File: training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc (hunk 3 above)
Comment (jeanchlopez) [reply training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc:3]: Done

File: training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc (hunk 4 above)
Comment (jeanchlopez) [reply training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc:4]: Done

File: training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc (hunk 7 above)
Comment (jeanchlopez) [reply training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc:42]: Done

File: training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc (hunk 10 above)
Comment (jeanchlopez) [reply training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc:87]: Addressed

Comment (jeanchlopez) [reply training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc:87]: Added a comment about WALK placement

File: training/modules/ocs4/pages/ocs4-additionalfeatures-override.adoc (hunk 11 above)
Comment (jeanchlopez) [reply training/modules/ocs4/pages/ocs4-additionalfeatures-override.adoc:19]: Changed the order of things in this module to make it more readbale.

**Review by mulbc — APPROVED** [review]

**Comment by mulbc** [comment]
I've added two commits, the first is to fix some small issues I found, the second is to show you my idea of moving all these docs into the nav bar. They will be hidden by default and you can expand them if you want to.

If you don't like this idea, just remove the last commit and we can keep it the way you planned it out.

File: training/modules/ocs4/pages/ocs4-additionalfeatures-override.adoc [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+a specific ConfigurationMap to be used by the operator during the deployment.
+
+[source,shell]
+----
+apiVersion: v1
+data:
+  config: |2
```
[the remark this answers is not in the document]
Comment (jeanchlopez) [reply training/modules/ocs4/pages/ocs4-additionalfeatures-override.adoc:35]: Hi Chris, this is what is documented in the engineering doc I once had access too. So kept it the same format and same typing

File: training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc (hunk 9 above)
Comment (jeanchlopez) [reply training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc:87]: Done

**Comment by jeanchlopez** [comment]
Works for me Chris if you want to place into the side bar.

Addressed Annette's comment yesterday in my local repo but did not push it yet. Some of your edit collide with my changes at this point. Will have to merge it all

**Review by jeanchlopez — COMMENTED** [review]
Looks good

File: training/modules/ocs4/pages/ocs4-additionalfeatures-override.adoc (hunk 5 above)
Comment (jeanchlopez) [reply training/modules/ocs4/pages/ocs4-additionalfeatures-override.adoc]: This has been addressed in the last commit I pushed.

File: training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc [hunk 14]
```diff
@@ -63,6 +48,23 @@ spec:
...
         storageClassName: {storageclass-ssd}
         volumeMode: Block
+----
+<1> The authorized values for the device types are `hdd`, `ssd` and `nvme`. Those device types will be used
+to assign a CRUSH device class within your underlying cluster. See the xref:ocs4-additionalfeatures-segregation.adoc[document on data segregation] for more information on this.
+
+== Hardware Configuration (HDD+SSD+NVMe)
+
+Your hardware configuration allows you to create partitions on an NVMe device for the OSD WAL,
+caete partitions on an SSD for OSD metadata and you want to assign a CRUSH Device Class as `ssd`
```
Comment (netzzer) [inline training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc:58]: caete -> create  AND ...you want to assign... -> ...you will also want to assign...

File: training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc [hunk 15]
```diff
@@ -101,8 +103,19 @@ to assign a CRUSH device class within your underlying cluster. See the xref:ocs4
 <2> Following the recommendations used for Red Hat Ceph Storage the only interest in separating the WAL placement
 is if the WAL can reside on a faster device than the device where the metadata lives.
 
+IMPORTANT: If you do not specify a `deviceType` value the OSD will auto-detect the type of device using
+the value contained in `/sys/block/{devicename}/queue/rotational`. If the file contains a `0` the device
+type assigned will be `ssd` but if the value is `1` teh device type assigned will be `hdd`.
```
Comment (netzzer) [inline training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc:108]: teh -> the

**Review by netzzer — APPROVED** [review]

File: training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc (hunk 14 above)
Comment (jeanchlopez) [reply training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc:58]: Done

File: training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+      spec:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: {size}
+        storageClassName: {storageclass}
+        volumeMode: Block
+----
+<1> The authorized values for the device types are `hdd`, `ssd` and `nvme`. Those device types will be used
```
[the remark this answers is not in the document]
Comment (jeanchlopez) [reply training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc:101]: There is no <3> note on this one and in yhe html generated I do see the <2> note. Not sure where you see a <3> and why you don't see the generated 2 note

File: training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc (hunk 15 above)
Comment (jeanchlopez) [reply training/modules/ocs4/pages/ocs4-additionalfeatures-devtype.adoc:108]: Done

**Pull request merged**